tools/bench-libs/src/main/java/org/enso/benchmarks/libs/SpecCollector.java

```diff
@@ -1,56 +1,90 @@
+  private final File rootDir;
   private final Context ctx;
+  private static final String allSuitesVarName = "all";
 
-  public SpecCollector(Context ctx) {
-    this.ctx = ctx;
+  public SpecCollector(File benchProjectDir, File languageHomeOverride) {
+    if (!benchProjectDir.exists() || !benchProjectDir.isDirectory()) {
+      throw new IllegalArgumentException(benchProjectDir + " is not a directory.");
+    }
+    this.rootDir = benchProjectDir;
+    this.ctx = Context.newBuilder()
+        .allowExperimentalOptions(true)
+        .allowIO(IOAccess.ALL)
+        .allowAllAccess(true)
+        .logHandler(new ByteArrayOutputStream())
+        .option(
+            "enso.projectRoot",
+            benchProjectDir.getAbsolutePath()
+        )
+        .option(
+            "enso.languageHomeOverride",
+            languageHomeOverride.getAbsolutePath()
+        )
+        .build();
   }
 
-  public Collection<BenchSuite> collectBenchSpecsFromDir(File rootDir) {
-    throw new UnsupportedOperationException("unimplemented");
+  public Collection<BenchSuite> collectAllBenchSpecs() {
+    try (Stream<Path> stream = Files.walk(rootDir.toPath())) {
+      return stream
+          .map(Path::toFile)
+          .filter(file -> file.isFile() && file.canRead() && file.getName().endsWith(".enso"))
+          .map(this::collectBenchSpecsFromSingleFile)
+          .filter(Objects::nonNull)
+          .collect(Collectors.toList());
+    } catch (IOException e) {
+      throw new IllegalStateException("Unreachable", e);
+    }
+
   }
 
   /**
-   *
-   * @param benchFile
+   * Collects benchmark specifications from a single file. Returns null if no specification
+   * is found in the file.
+   * @param benchFile Path to the source file.
    * @return null if there are no benchmark specs in the file.
    */
-  public BenchSuite collectBenchSpecsFromSingleFile(File benchFile) {
-    Objects.requireNonNull(benchFile);
-    assert benchFile.exists();
-    assert benchFile.canRead();
+  private BenchSuite collectBenchSpecsFromSingleFile(File benchFile) {
     Source source;
     try {
       source = Source.newBuilder("enso", benchFile).build();
     } catch (IOException e) {
       throw new IllegalStateException("Unreachable", e);
     }
     Value module = ctx.eval(source);
-    BenchSuite benchSuite = null;
-    Value associatedType = module.invokeMember(Module.GET_ASSOCIATED_TYPE);
-    // TODO: Try to
-    try {
-      benchSuite = module
-          .invokeMember(Module.EVAL_EXPRESSION, "all")
-          .as(BenchSuite.class);
-    } catch (PolyglotException e) {
-      // TODO: Handle this properly
-      throw new IllegalStateException("Unreachable", e);
+    Value moduleType = module.invokeMember(Module.GET_ASSOCIATED_TYPE);
+    Value allSuitesVar = module.invokeMember(Module.GET_METHOD, moduleType, allSuitesVarName);
+    if (!allSuitesVar.isNull()) {
+      try {
+        return module
+            .invokeMember(Module.EVAL_EXPRESSION, "all")
```

Comment:
The name doesn't have to be `all` - I used it, but we can choose anything more descriptive `bench`, `all_benchmarks`, etc.

Reply:
This is already implemented, as I have noted in some of my previous comments.